dask/array/core.py

```diff
@@ -1342,8 +1342,14 @@ def _vindex(self, key):
                 "vindex does not support indexing with None (np.newaxis), "
                 "got {}".format(key))
         if all(isinstance(k, slice) for k in key):
+            slices = (k.indices(num_dim) for k, num_dim in zip(key, self.shape))
+            identity_slices = ((0, num_in_dim, 1)   # i.e, x[:]
+                               for num_in_dim in self.shape)
+            if all(slice_ == identity for slice_, identity in zip(slices, identity_slices)):
+                return self
```

Comment:
Perhaps inline `identity_slices` here?

    if all(slc == slice(0, d) for slc, d in zip(slices, self.shape)):
       ...

This might reduce cognitive load from future readers a bit, rather than have to cross check against the variable definition above.

Reply:
Nice. That helps makes the code simpler.